pkg/runtime/engine/config/parser.go

```diff
@@ -17,147 +17,156 @@ import (
 	"text/template"
 )
 
-// Parser parses configuration found in the file with the specified path.
-func Parser(filepath string, f interface{}) error {
-	var err error
-	var c *os.File
-	var b []byte
-	directives := make(map[string][]string)
+// Directives represents the configuration directives type
+// holding directives mapped to their respective values.
+type Directives map[string][]string
 
-	if filepath != "" {
-		c, err = os.Open(filepath)
-		if err != nil {
-			return err
-		}
-		b, err = ioutil.ReadAll(c)
-		if err != nil {
-			return err
-		}
+// ParseData parses data and returns a directive map with
+// associated value.
+func ParseData(reader io.Reader) (Directives, error) {
+	var buf bytes.Buffer
 
-		c.Close()
+	if reader != nil {
+		if _, err := io.Copy(&buf, reader); err != nil {
+			return nil, err
+		}
 	}
 
 	r, err := regexp.Compile(`(?m)^\s*([a-zA-Z _]+)\s*=\s*(.*)$`)
```

Comment:
Can be defined as a package var with `regexp.MustCompile`. Just a suggestion, this way is also fine taking into account when `Parser` is called.